mesh/v1alpha1/config.proto

```
@@ -505,7 +505,97 @@ message MeshConfig {
+  message ExtensionProvider {
+    // REQUIRED. A unique name identifying the extension provider.
+    string name = 1;
+
+    // REQUIRED. Specifies detailed configuration for the extension provider.
+    oneof provider {
+      // Configures an external authorizer that implements the Envoy ext_authz filter authorization check service using the HTTP API.
+      EnvoyExternalAuthorizationHttpProvider envoy_ext_authz_http = 2;
+      // Configures an external authorizer that implements the Envoy ext_authz filter authorization check service using the gRPC API.
+      EnvoyExternalAuthorizationGrpcProvider envoy_ext_authz_grpc = 3;
+    }
+
+    // $hide_from_docs
+    message EnvoyExternalAuthorizationHttpProvider {
+      // REQUIRED. Specifies the service that implements the Envoy ext_authz HTTP authorization service.
+      // The format is "[<Namespace>/]<Service>". The <Namespace> will have the default value "istio-system" if it is
+      // omitted. The <Service> is the name of the service object (k8s service or ServiceEntry).
+      // Example: "foo/my-ext-authz" or "my-ext-authz".
+      string service = 1;
+
+      // Specifies the port of the service. Can be omitted if it can be decided unambiguously for the service (e.g. the
+      // service has defined only 1 port).
+      uint32 port = 2;
+
+      // Sets a prefix to the value of authorization request header *Path*.
+      // For example, setting this to "/check" for an original user request at path "/admin" will cause the
+      // authorization check request to be sent to the authorization service at the path "/check/admin" instead of "/admin".
+      string path_prefix = 3;
+
+      // If true, the user request will be allowed even if the communication with the authorization service has failed,
+      // or if the authorization service has returned a HTTP 5xx error.
+      // Default is false and the request will be rejected with "Forbidden" response.
+      bool fail_open = 4;
```

@costinm 
@smawson 

should this setting be hoisted a level? Isn't this generic for all possible providers

---

I'm not sure, would a telemetry provider have fail open vs. fail closed?

---

Chatted with Louis and since it doesn't actually save our users any typing we can leave this down here on the hope that extension providers is useful beyond authorization.

---

yeah, I also prefer to leave it to the specific type as some other extension provider may doesn't need or have a different interpretation or default value for the `fail_open` concept.